### PR TITLE
Add `c_call` and `c_return` trace points for `struct_aref` and `struct_aset`.

### PR DESCRIPTION
**Pull request by ioquatix**

### CODE REVIEW

File: vm_insnhelper.c [hunk 1]
```diff
@@ -3539,8 +3539,17 @@ static VALUE
 vm_call_opt_struct_aref(rb_execution_context_t *ec, rb_control_frame_t *reg_cfp, struct rb_calling_info *calling)
 {
     RB_DEBUG_COUNTER_INC(ccf_opt_struct_aref);
+    const struct rb_callinfo *ci = calling->ci;
```
Comment (ko1) [inline vm_insnhelper.c:3542]: don't introduce ci/cc lvars and use `calling->ci`, `calling->cc` directly to save non-TP enable time (99.999...%)

Comment (ko1) [reply vm_insnhelper.c:3542]: or `if (UNLIKELY(ruby_vm_event_flags & RUBY_EVENT_C_CALL))` at first.

```ruby
if (LIKELY(!ruby_vm_event_flags & RUBY_EVENT_C_CALL)) {
  ret = vm_call_opt_struct_aref0(ec, calling);
}
else {
  ci = ...
  cc = ...
  EXEC_EVENT_HOOK(ec, RUBY_EVENT_C_CALL, ...
  ret = vm_call_opt_struct_aref0(ec, calling);
  RUBY_EVENT_C_RETURN))
        EXEC_EVENT_HOOK(ec, RUBY_EVENT_C_RETURN
}
```

it also reduces branches.

Comment (ioquatix) [reply vm_insnhelper.c]: Okay done :)

**Comment by ioquatix** [comment]
@ko1 I'll close this PR.

**Pull request closed without merging**